### PR TITLE
Docs: elaborate on parse phases comments

### PR DESCRIPTION
**Pull request by agoose77**
Whilst reading the comments for the parsing logic, I found it helpful to add a few more lines to explicitly clarify what happens.

### CODE REVIEW

File: news/docs-elaborate-parse-phases.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+**Added:**
+
+* Docs to the execer parse phase.
+
```
Comment (jnoortheen) [inline news/docs-elaborate-parse-phases.rst:4]: Does this need a news item? Also this wont get pushed in the html docs that the user normally sees.

**Comment by jnoortheen** [comment]
@agoose77 docs are dlgood. Can you please check the failing qa checks?

File: news/docs-elaborate-parse-phases.rst (hunk 1 above)
Comment (agoose77) [reply news/docs-elaborate-parse-phases.rst:4]: I'm not sure - I just added it because it was required for the CI?

Comment (jnoortheen) [reply news/docs-elaborate-parse-phases.rst:4]: You can remove remove it in this case.

**Comment by jnoortheen** [comment]
@agoose77 sorry for pinging again. If it is any help, removing the news item will make the qa checks pass.

**Comment by jnoortheen** [comment]
Thanks @agoose77 ! I went ahead and merging this PR so that other PRs can have more visibility.

**Review by jnoortheen — APPROVED** [review]

**Comment by agoose77** [comment]
Thanks a bunch! I'm sorry for not getting to this sooner - I've had a lot
going on my end the last couple of weeks.

**Pull request merged**